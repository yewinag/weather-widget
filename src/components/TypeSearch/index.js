import React, { Component } from 'react';
import styled from 'styled-components';
 
class TypeSearch extends Component {
    render() { 
        return (
            <div>
                <SearchInput 
                    placeholder="Type city name" 
                    onChange={(e)=> this.props.onChange(e.target.value)} 
                />
                <SearchButton onClick={()=> this.props.onSearch()}>
                    Search
                </SearchButton>
            </div>
        );
    }
}
 
export default TypeSearch;

const SearchInput = styled.input`
    font-size: 18px;
    padding: 10px;
    color: #000;
`;

const SearchButton = styled.button`
    padding: 10px;
    color: '#000';
    backgroundColor: palevioletred;
`