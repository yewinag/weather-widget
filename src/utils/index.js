
import { applyMiddleware, createStore } from 'redux';
import rootReducer from '../reducers';
import { middlewares } from '../store';

export const testStore = (initialState) => {
    const createStoreWithMiddleware = applyMiddleware(...middlewares)(createStore);
    return createStoreWithMiddleware(rootReducer, initialState);
};