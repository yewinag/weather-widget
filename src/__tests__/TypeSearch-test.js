import React from 'react';
import { Provider } from 'react-redux';
import { testStore } from '../utils';
import TypeSearch from '../components/TypeSearch';

import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Enzyme from 'enzyme';

Enzyme.configure({ adapter: new Adapter() });

describe('Type search', () => {
    let store;
    beforeEach(() => {
        store = testStore();
    });   
    
    it('render type search', () => {
        const wrapper = shallow(
            <Provider store={store}>
                <TypeSearch />                
            </Provider>
        )
        expect(wrapper.exists()).toBe(true);
        expect(wrapper).toMatchSnapshot();
    });
});