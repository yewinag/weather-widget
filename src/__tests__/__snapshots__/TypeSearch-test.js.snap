// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Type search render type search 1`] = `ShallowWrapper {}`;
